### PR TITLE
Update docs pages

### PR DESCRIPTION
**Pull request by glenn-jocher**
@kalenmike @hassaanfarooq01 

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
The `ultralytics/hub-sdk` GitHub PR introduces improvements to documentation and configuration settings.

### 📊 Key Changes
- Removed `double-quote-string-fixer` config in `.pre-commit-config.yaml` for more flexible string formatting.
- Enhanced documentation for dataset, model, project management, and overall SDK usage with descriptive content and structure.
- Enriched Quickstart Guide with detailed setup instructions and contextual information.

### 🎯 Purpose & Impact
- Improve developer experience by allowing both single and double-quoted strings in code.
- Provide users with clear, comprehensive guidance on managing datasets, models, and projects using the Ultralytics HUB-SDK.
- Help new users efficiently install and initialize the SDK, increasing accessibility and ease of use.
- Simplified and contextual documentation encourages broader SDK adoption and smoother onboarding for users with different expertise levels.

### CODE REVIEW

File: docs/model.md [hunk 1]
```diff
@@ -1,119 +1,133 @@
...
 model = client.model("<Model ID>")
-print(model.data)
+print(model.data)  # Outputs the model's metadata and configuration
 ```
 
-### Get Project and Dataset
+## Access Project and Dataset Prerequisites
 
-This code snippet ensures the presence of both a project and a dataset by initializing their objects using their respective IDs. In case either of these IDs is not available, it raises an exception, as both are essential prerequisites for the creation of a model.
+Prior to model creation or training, ensuring the presence of both a project and a dataset is crucial. This straightforward code snippet helps you verify these components are available by initializing their objects. If either ID is missing, an exception is a clear indicator you need to address this before proceeding.
```
Comment (hassaanfarooq01) [inline docs/model.md:22]: Little update with description. to add up it's not required. and not raise the exception.  
```suggestion
Prior to model creation or training, ensuring the presence of both a project and a dataset is crucial. This straightforward code snippet helps you verify these components are available by initializing their objects. While utilizing a project and dataset for organizing model training is beneficial, it's important to note that it is not mandatory. If either ID is missing, the object data (`project.data`, `dataset.data`) will  be empty.
```

File: docs/model.md [hunk 2]
```diff
@@ -1,119 +1,133 @@
...
+        "patience": "5"             # Stop training if validation doesn't improve
     }
 }
 model = client.model()
-model.create_model(data)
+model.create_model(data)  # Creates the model with your specified details
 ```
 
-### Update Model
+## Update an Existing Model's Metadata
```
Comment (hassaanfarooq01) [inline docs/model.md:51]:
```suggestion
## Update an Existing Model's Metadata or Config
```

File: docs/model.md [hunk 3]
```diff
@@ -1,119 +1,133 @@
...
-### Delete Model
+## Delete a Model Safely
 
-This function lets you erase a particular model by giving its ID. Use it carefully, as it permanently erases the model and its linked data.
+Deleting a model is irreversible, so this function should be used with caution. When you're sure you want to remove a model from the system, the following command will permanently delete the specified model, along with all its associated data.
 
 ```python
 model = client.model("Model ID")
-model.delete()
+model.delete()  # Permanently deletes the specified model
```
Comment (hassaanfarooq01) [inline docs/model.md:66]:
```suggestion
model.delete(hard=True)  # Permanently deletes the specified model
```

File: docs/model.md [hunk 4]
```diff
@@ -1,119 +1,133 @@
...
 ```
```
Comment (hassaanfarooq01) [inline docs/model.md:68]:
```suggestion
By default, the delete method performs a soft delete, marking the model as inactive without removing it permanently. If you want to perform a hard delete and remove the model along with its associated data permanently, you can pass the argument hard=True as shown in the example above. Exercise caution when using the hard delete option, as it is irreversible and results in the complete removal of the specified model from the system.

```

**Comment by glenn-jocher** [comment]
@hassaanfarooq01 hey bud, just go ahead and directly take over and modify this PR, it's yours and @kalenmike's to do what you want with. Faster and simpler than suggesting changes.

**Comment by glenn-jocher** [comment]
Once we get this cleaned up and merged then I can publish these by merging  https://github.com/ultralytics/ultralytics/pull/7775

**Pull request merged**